cluster/src/main/scala/io/snappydata/impl/LeadImpl.scala

```diff
@@ -377,13 +381,19 @@ class LeadImpl extends ServerImpl with Lead
         val primaryLeaderLock = new DistributedMemberLock(dls,
           LOCK_SERVICE_NAME, DistributedMemberLock.NON_EXPIRING_LEASE,
           DistributedMemberLock.LockReentryPolicy.PREVENT_SILENTLY)
-
         val startStatus = primaryLeaderLock.tryLock()
         // noinspection SimplifyBooleanMatch
         startStatus match {
           case true =>
             logInfo("Primary lead lock acquired.")
 
+            // store unique cluster id to metadataCmdRgn
+            var clusterUuid = Misc.getMemStore.getMetadataCmdRgn.get("__ClusterID__")
+            if (clusterUuid == null) {
+              val region = Misc.getMemStore.getMetadataCmdRgn
+              region.put("__ClusterID__", UUID.randomUUID().toString)
```

Comment:
As I said earlier, I feel user defined cluster name is much more useful than random uuid. If there are multiple systems then UUID will give no help to the user to identify a particular cluster. It should not be difficult to add.

@jramnara, @sudsmenon  please comment.

Reply:
It would be surely taken up for next release . Jira ticket for this task is https://jira.snappydata.io/browse/SNAP-3230